syncopy/tests/test_selectdata.py

```diff
@@ -9,749 +9,557 @@
+    nSpikes = 20
+    T_max = 2 * nSpikes   # in samples, not seconds!
+    nSamples = T_max / nTrials
+    rng = np.random.default_rng(42)
+
+    data = np.vstack([np.sort(rng.choice(range(T_max), size=nSpikes)),
+                      rng.choice(np.arange(0, nChannels), size=nSpikes),
+                      rng.choice(nChannels // 2, size=nSpikes)]).T
+
+    trldef = np.vstack([np.arange(0, T_max, nSamples),
+                        np.arange(0, T_max, nSamples) + nSamples,
+                        np.ones(nTrials) * -2]).T
+
+    spike_data = spy.SpikeData(data=data,
+                               samplerate=samplerate,
+                               trialdefinition=trldef)
+
+    def test_spike_selection(self):
+
+        """
+        Create a typical selection and check that the returned data is correct
+        """
+
+        selection = {'trials': [2, 4],
+                     'channel': [6, 2],
+                     'unit': [0, 3],
+                     'latency': [-1, 4]}
+        res = self.spike_data.selectdata(selection)
+
+        # hand pick selection from the arrays
+        dat_arr = self.spike_data.data
+
+        # these are trial intervals in sample indices!
+        trial2 = self.spike_data.trialdefinition[2, :2]
+        trial4 = self.spike_data.trialdefinition[4, :2]
+
+        # create boolean mask for trials [2, 4]
+        bm = (dat_arr[:, 0] >= trial2[0]) & (dat_arr[:, 0] <= trial2[1])
+        bm = bm | (dat_arr[:, 0] >= trial4[0]) & (dat_arr[:, 0] <= trial4[1])
+
+        # add channels [6, 2]
+        bm = bm & ((dat_arr[:, 1] == 6) | (dat_arr[:, 1] == 2))
+
+        # units [0, 3]
+        bm = bm & ((dat_arr[:, 2] == 0) | (dat_arr[:, 2] == 3))
+
+        # latency [-1, 4]
+        # to index all trials at once
+        time_vec = np.concatenate([t for t in self.spike_data.time])
+        bm = bm & ((time_vec >= -1) & (time_vec <= 4))
+
+        # finally compare to selection result
+        assert np.all(dat_arr[bm] == res.data[()])
+
+    def test_spike_valid(self):
+
+        """
+        Instantiate Selector class and check only its attributes, the idx
+        used by `_preview_trial` in the end
+        """
+
+        # each selection test is a 2-tuple: (selection kwargs, dict with same kws and the idx "solutions")
+        valid_selections = [
+            (
+                # units get apparently indexed on a per trial basis
+                {'trials': np.arange(1, 4), 'channel': ['channel03', 'channel01'], 'unit': [2, 0]},
+                {'trials': [1, 2, 3], 'channel': [2, 0], 'unit': [[], [], [1, 5]]},
+            ),
+            # 2nd selection
+            (
+                # time/latency idx can be mixed lists and slices O.0
+                # and channel 'all' selections can still be effectively subsets..
+                {'trials': [0, 4], 'latency': [0, 3], 'channel': 'all'},
+                {'trials': [0, 4], 'latency': [slice(0, 4, 1), [1]], 'channel': [1, 2, 3, 5, 9]},
```

Comment:
This `[1]` seems to have failed in the tests, I also don't get what it's the solution for.

Reply:
Ah okay I see, it was supposed to be the valid part of trial 4 but now that the sampling rate is different the output is different. Right?

Reply:
hehe.. right :)